src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -188,7 +188,18 @@ CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanResultCallback resultCal
     mWiFiState          = WIFI_STATE_SCANNING;
     mSsidFound          = false;
 
-    if (0 != net_mgmt(NET_REQUEST_WIFI_SCAN, mNetIf, NULL, 0))
+    wifi_scan_params * scanParamsFinal{ nullptr };
+
+    if (!ssid.empty())
+    {
+        /* We must assume that the ssid is handled as a NULL-terminated string.
+           Note that the mScanSsidBuffer is initialized with zeros. */
+        VerifyOrReturnError(ssid.size() < sizeof(mScanParams), CHIP_ERROR_INVALID_ARGUMENT);
+        memcpy(mScanSsidBuffer, ssid.data(), ssid.size());
```

Comment:
Can you add null-terminator here? Thinking about the case in which we first scan for "my-network-1" and then some shorter "my-network".

Reply:
Valid point, thanks.